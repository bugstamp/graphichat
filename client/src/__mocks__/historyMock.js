export const historyPushMock = jest.fn(str => str);

const historyMock = {
  push: historyPushMock,
};

export default historyMock;
