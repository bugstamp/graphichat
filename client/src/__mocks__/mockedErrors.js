import { createError } from 'apollo-errors';

export const AuthenticationError = createError('AuthenticationError', {
  message: 'The user is not authorized',
});

export const BadInputError = createError('BadInputError', {
  message: 'The provided value is invalid',
});

export const ForbiddenError = createError('ForbiddenError', {
  message: 'Access is denied',
});

export const ServerError = createError('ServerError', {
  message: 'Server Error',
});

export class GraphQLErrors extends Error {
  constructor(graphQLErrors) {
    super();
    this.graphQLErrors = graphQLErrors;
    this.name = 'GraphQLErrors';
  }
}
