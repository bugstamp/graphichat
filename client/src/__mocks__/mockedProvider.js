import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { InMemoryCache } from 'apollo-boost';
import { MockedProvider } from '@apollo/react-testing';
import { MuiThemeProvider } from '@material-ui/core/styles';
import { ThemeProvider } from 'styled-components';
import { mount } from 'enzyme';

import { initialState } from '../apollo/cache';
import { theme } from '../styles';

// eslint-disable-next-line
const TestProvider = ({ children, mocks, cache, history }) => (
  <MuiThemeProvider theme={theme}>
    <ThemeProvider theme={theme}>
      <MockedProvider
        mocks={mocks}
        cache={cache}
        addTypename={false}
      >
        <Router history={history}>
          {children}
        </Router>
      </MockedProvider>
    </ThemeProvider>
  </MuiThemeProvider>
);

export const mountMockedProvider = (children, mocks = [], {
  history = createMemoryHistory(),
  state = null,
} = {}) => {
  const cache = new InMemoryCache({ addTypename: false });
  cache.writeData({ data: state || initialState });

  return mount((
    <TestProvider
      mocks={mocks}
      cache={cache}
      history={history}
    >
      {children}
    </TestProvider>
  ));
};

export default TestProvider;
