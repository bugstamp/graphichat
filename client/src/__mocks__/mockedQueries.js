import gql from '../gql';
import {
  avatar,
  me,
  user,
  userActivityUpdate,
  contact,
  chat,
} from './mockedQueryData';

const {
  CHECK_SESSION_EXPIRATION,
  GET_INITIAL_DATA,
  UPDATE_USER,
  UPLOAD_AVATAR,
  USER_UPDATE_SUBSCRIPTION,
  USER_ACTIVITY_SUBSCRIPTION,
  CHAT_CREATED_SUBSCRIPTION,
  SIGN_OUT,
} = gql;

export const checkSessionExpirationMock = {
  request: {
    query: CHECK_SESSION_EXPIRATION,
    variables: {},
  },
  result: {
    data: {
      me,
      myContacts: [],
      myChats: [],
    },
  },
};
export const getInitialDataMock = {
  request: {
    query: GET_INITIAL_DATA,
    variables: {},
  },
  result: {
    data: {
      me,
      myContacts: [],
      myChats: [],
    },
  },
};
export const updateUserMock = {
  request: {
    query: UPDATE_USER,
    variables: {
      field: 'username',
      value: 'user',
    },
  },
  result: {
    data: {
      updateUser: me,
    },
  },
};
export const uploadAvatarMock = {
  request: {
    query: UPLOAD_AVATAR,
    variables: {
      file: {},
    },
  },
  result: {
    data: {
      uploadAvatar: avatar,
    },
  },
};
export const userUpdateSubscriptionMock = {
  request: {
    query: USER_UPDATE_SUBSCRIPTION,
    variables: {},
  },
  result: {
    data: {
      userUpdated: me,
    },
  },
};
export const userActivitySubscriptionMock = {
  request: {
    query: USER_ACTIVITY_SUBSCRIPTION,
    variables: {},
  },
  result: {
    data: {
      userActivityUpdated: userActivityUpdate,
    },
  },
};
export const chatCreatedSubscriptionMock = {
  request: {
    query: CHAT_CREATED_SUBSCRIPTION,
    variables: {},
  },
  result: {
    data: {
      chatCreated: {
        contact,
        chat,
      },
    },
  },
};
export const signOutMock = {
  request: {
    query: SIGN_OUT,
    variables: {
      userId: me.id,
    },
  },
  result: {
    data: true,
  },
};
