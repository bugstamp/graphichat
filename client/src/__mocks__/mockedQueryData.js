export const avatar = {
  sm: '',
  md: '',
  __typename: 'UserAvatar',
};

export const uploadAvatar = {
  ...avatar,
  sm: 'smAvatar',
  md: 'mdAvatar',
};

export const me = {
  id: 1,
  avatar,
  username: 'me',
  displayName: 'Me Test',
  firstName: 'Me',
  lastName: 'Test',
  status: 'ONLINE',
  lastDate: Date.now(),
  __typename: 'User',
};

export const user = {
  id: 2,
  avatar,
  username: 'user',
  displayName: 'User Test',
  firstName: 'User',
  lastName: 'Test',
  status: 'OFFLINE',
  lastDate: Date.now(),
  __typename: 'User',
};

export const userActivityUpdate = {
  userId: 2,
  status: 'ONLINE',
  lastDate: Date.now(),
};

export const contact = {
  id: 11,
  chatId: 1,
  userInfo: user,
};

export const message = {
  id: 1,
  senderId: 1,
  type: 'text',
  content: 'yo!',
  time: Date.now(),
  edited: false,
  seen: false,
};

export const chat = {
  id: 1,
  messages: [message],
};
