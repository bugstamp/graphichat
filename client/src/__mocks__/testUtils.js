// fix bug with react-facebook-login lib
export const fixReactFacebookLoginBug = () => {
  const script = document.createElement('script');
  script.type = 'text/javascript';
  document.getElementsByTagName('head')[0].appendChild(script);
};
