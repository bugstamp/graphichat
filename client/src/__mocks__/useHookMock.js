import React from 'react';

const useHookMock = (hook, returnValue) => {
  return jest.spyOn(React, hook).mockImplementation(() => returnValue);
};

export default useHookMock;
