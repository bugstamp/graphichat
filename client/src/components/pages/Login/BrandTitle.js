import React from 'react';

import Typography from '@material-ui/core/Typography';
import useMediaQuery from '@material-ui/core/useMediaQuery';

import Logo from '../../common/Logo';
import { BrandTitleWrapper } from './styled';

const getLogoParams = (downMd, downSm) => {
  if (downSm) {
    return [35, 'h3'];
  }
  if (downMd) {
    return [35, 'h2'];
  }
  return [60, 'h1'];
};

const BrandTitle = () => {
  const downMd = useMediaQuery(theme => theme.breakpoints.down('md'));
  const downSm = useMediaQuery(theme => theme.breakpoints.down('sm'));
  const [logoSize, typographyVariant] = getLogoParams(downMd, downSm);

  return (
    <BrandTitleWrapper>
      <Logo size={logoSize} />
      <Typography
        component="p"
        variant={typographyVariant}
        align="center"
      >
        GraphiChat
      </Typography>
    </BrandTitleWrapper>
  );
};

export default BrandTitle;
