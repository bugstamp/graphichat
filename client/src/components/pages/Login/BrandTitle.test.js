import React from 'react';
import { shallow } from 'enzyme';
import Typography from '@material-ui/core/Typography';

import BrandTitle from './BrandTitle';

import { mountMockedProvider } from '../../../__mocks__/mockedProvider';
import mockedUseMediaQuery from '../../../__mocks__/@material-ui/core/useMediaQuery';

describe('LoginPresentation', () => {
  const mountWrapper = () => mountMockedProvider(<BrandTitle />);

  test('should match to snapshot', () => {
    const wrapper = shallow(<BrandTitle />);

    expect(wrapper).toMatchSnapshot();
  });
  test('mount should be without errors', () => {
    const wrapper = mountWrapper();

    expect(wrapper.find('BrandTitle')).toHaveLength(1);
  });
  test('should have Logo', () => {
    const wrapper = mountWrapper();

    expect(wrapper.find('Logo')).toHaveLength(1);
  });
  test('logoSize and typographyVariant definition', () => {
    mockedUseMediaQuery.mockImplementation(() => false);
    const wrapper = mountWrapper();

    expect(wrapper.find('Logo').prop('size')).toBe(60);
    expect(wrapper.find(Typography).prop('variant')).toBe('h1');

    wrapper.unmount();
    mockedUseMediaQuery.mockImplementation(() => true);
    wrapper.mount();

    expect(wrapper.find('Logo').prop('size')).toBe(35);
    expect(wrapper.find(Typography).prop('variant')).toBe('h3');
  });
  test('typography props', () => {
    mockedUseMediaQuery.mockImplementation(() => false);
    const wrapper = mountWrapper();

    expect(wrapper.find(Typography).props()).toMatchObject({
      variant: 'h1',
      component: 'p',
      align: 'center',
      children: 'GraphiChat',
    });
  });
});
