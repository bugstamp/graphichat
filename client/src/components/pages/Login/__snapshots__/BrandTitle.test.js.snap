// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPresentation should match to snapshot 1`] = `
<styled__BrandTitleWrapper>
  <Logo
    size={35}
  />
  <WithStyles(ForwardRef(Typography))
    align="center"
    component="p"
    variant="h3"
  >
    GraphiChat
  </WithStyles(ForwardRef(Typography))>
</styled__BrandTitleWrapper>
`;
